src/Models/Board.js

```diff
@@ -143,33 +143,100 @@ export default class Board{
                 return new Cheetah(x,y);
         }
         return null;
+    }
+
+    generateBasicPieces(x,y,player_){
+        for(; x < this.COL_SIZE - 2; x++){
+            this.grid[x][y].Occupant = new Chicken(x,y);
+            this.grid[x][y].Occupant.Owner = player_;
+        }
+    }
 
+    generateSpecialPieces(x,y,player_){
+        console.log(x)
+        if(y == 0){
```

Comment:
I'd rather have the conditions be checked outside of the function. 
Function should be versatile.
If statements should be in createnewGrid()
 --- I prefer modularity over straight time optimization

Reply:
The special pieces need to be generated mirrored to the opponent's so I made it so that the pieces are generated in opposite order through either incrementing or decrementing the index. I don't see a way I could do this without checking a condition within the function without it looking weird. Even if I determine start and end indices I still need to determine whether to increment or decrement which I could do by passing in an argument either 1 or -1 to act as the iterator. So it's either going to look like how I had it before or

generateSpecialPieces(start,end,y,iterator,player){
  for(var x = start; x != end; x += iterator){
    if(x == 2){ // generate cheetah
      this.grid[x][y].Occupant = new Cheetah(x,y);
      this.grid[x][y].Occupant.Owner = player;
    }
    else if(... // generate other pieces)
    .
    .
    .
}

By nature, this function isn't intended to be versatile since it will only be called in two special cases. Would you still prefer it this way? It's going to result in half the amount of code but some overhead for the additional arguments being pushed (not that it will have a noticeable effect).

Reply:
![setup](https://user-images.githubusercontent.com/23038573/68058630-dfc52f00-fcb6-11e9-8066-6e3524ab16b8.png)
They're exact mirrors of each other. 
Why is there a need to switch from increment and decrement?

Reply:
When I looked up the layout of the pieces the queen and king were opposite.